### PR TITLE
Backend peer

### PR DESCRIPTION
**Pull request by vojtechsimetka**

### CODE REVIEW

**Review by mortelli — APPROVED** [review]

**Review by nonsense — APPROVED** [review]

File: contracts/swap/swap.go [hunk 1]
```diff
@@ -92,3 +93,8 @@ func (s *Swap) ContractParams() *Params {
 func (s *Swap) SubmitChequeBeneficiary(opts *bind.TransactOpts, serial *big.Int, amount *big.Int, timeout *big.Int, ownerSig []byte) (*types.Transaction, error) {
 	return s.Instance.SubmitChequeBeneficiary(opts, serial, amount, timeout, ownerSig)
 }
+
+func (s *Swap) InstanceAt(address common.Address, backend bind.ContractBackend) *contract.SimpleSwap {
```
Comment (mortelli) [inline contracts/swap/swap.go:97]: This could probably be called `CreateContractReference` or similar, for clarity.

File: contracts/swap/swap.go [hunk 2]
```diff
@@ -92,3 +93,8 @@ func (s *Swap) ContractParams() *Params {
...
+	ref, _ := contract.NewSimpleSwap(address, backend)
```
Comment (mortelli) [inline contracts/swap/swap.go:98]: Missing error-handling.

File: swap/protocol.go [hunk 3]
```diff
@@ -53,23 +49,26 @@ func (s *Swap) APIs() []rpc.API {
...
 
-func (ss *Service) run(p *p2p.Peer, rw p2p.MsgReadWriter) error {
+func (s *Swap) run(p *p2p.Peer, rw p2p.MsgReadWriter) error {
 	protoPeer := protocols.NewPeer(p, rw, Spec)
-	swapPeer := NewPeer(protoPeer, ss.swap)
+	swapPeer := NewPeer(protoPeer, s, s.backend)
 	return swapPeer.Run(swapPeer.handleMsg)
 }
+
+type PublicAPI struct {
```
Comment (mortelli) [inline swap/protocol.go:73]: Is this used somewhere?

Comment (nonsense) [reply swap/protocol.go:73]: This is supposed to hold the handlers for the public API endpoints, similar to `Inspector` struct, within the `bzz` namespace. I am not sure if it is `required`, or `optional`, but this is why we added it.

**Pull request merged**